Pod/API/Internal/Parse.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  Parse.swift
+//  ElJayKit
+//
+//  Created by Isaac Greenspan on 12/12/15.
+//  Copyright © 2015 Isaac Greenspan. All rights reserved.
+//
+
+import Foundation
+import AlamofireXMLRPC
+
+protocol XMLRPCDeserializable {
+    init(xmlrpcNode value: XMLRPCNode) throws
+}
+
+extension API {
+    enum ParseError: ErrorType {
+        case BadStructure
+        case BadStructureInArray
+    }
+    
+    /**
+     Internal XMLRPC call handler with single-item response parsing.
+     
+     - parameter method:              The method to call
+     - parameter parameters:          The parameters of the call, if any
+     - parameter responseType:        The type of item expected in the response
+     - parameter responsePostProcess: A function to apply to the parsed response before invoking the callback
+     - parameter callback:            The callback to execute with the result of the call
+     */
+    func execute<ResponseType: XMLRPCDeserializable>(
+        method: Method,
+        parameters: [String: Any]? = nil,
+        responseType: ResponseType.Type,
+        responsePostProcess: (ResponseType) -> (ResponseType?) = { $0 },
+        callback: Of<ResponseType>.callback) {
+            self.execute(method, parameters: parameters, callback: { result in
+                // Pass along obvious errors.
+                guard let value = result.value as? XMLRPCNode where result.isSuccess else {
+                    callback(.Failure(result.error!))
+                    return
+                }
+                
+                // Error if we didn't get a well-formed response structure.
+                guard let parsedValue = try? ResponseType(xmlrpcNode: value) else {
+                    callback(.Failure(.Parsing))
+                    return
+                }
+                
+                // Apply post-processing.
+                guard let postProcessedValue = responsePostProcess(parsedValue) else {
+                    callback(.Failure(.Parsing))
```

Comment:
It probably should be its own separate error...  Unless maybe `reponsePostProcess` should be able to `throw` and then use that error... except that it'd have to be an `API.Error` because I couldn't figure out how to leave the callback error type fully generically `ErrorType`.